src/zones/erldns_zone_cache.erl

```diff
@@ -747,8 +747,23 @@ is_record_name_in_zone_with_wildcard(ZoneLabels, QLabels) ->
     is_record_name_in_zone_traverse_wildcard(ZoneLabels, WildcardPath, Parent).
 
 is_record_name_in_zone_traverse_wildcard(ZoneLabels, Path, ParentPath) ->
-    0 =/= pattern_zone_dname_count(ZoneLabels, Path) orelse
-        is_record_name_in_zone_with_wildcard(ZoneLabels, ParentPath).
+    case pattern_zone_dname_count(ZoneLabels, Path) of
+        0 ->
+            UseCompliantENT = rfc_compliant_ent(),
+            traverse_wildcard(UseCompliantENT, ZoneLabels, ParentPath);
+        _ ->
+            true
+    end.
+
+traverse_wildcard(true = _UseCompliantENT, ZoneLabels, ParentPath) ->
```

Comment:
I see that most (or maybe all) of the helpers that return a boolean in this module start with `is_`. Could you come up with a name that would follow this convention? I think it could be misleading without it.